akka-remote/src/main/scala/akka/remote/artery/Codecs.scala

```diff
@@ -372,8 +384,17 @@ private[remote] class Decoder(
                     attemptsLeft - 1,
                     recipientPath, inboundEnvelope), retryResolveRemoteDeployedRecipientInterval)
                 else {
+                  // No more attempts left. If the retried resolve isn't successful the ref is banned and
+                  // we will not do the delayed retry resolve again. The reason for that is
+                  // if many messages are sent to such dead refs the resolve process will slow
+                  // down other messages.
+                  if (bannedRemoteDeployedActorRefs.size >= 100) {
```

Comment:
It would be nicer to hide this in the datastructure than to spread to the points at which we do writes.